include/tvm/tir/expr.h

```diff
@@ -675,7 +679,8 @@ class BufferLoadNode : public PrimExprNode {
  */
 class BufferLoad : public PrimExpr {
  public:
-  TVM_DLL explicit BufferLoad(Buffer buffer, Array<PrimExpr> indices, Span span = Span());
+  TVM_DLL explicit BufferLoad(Buffer buffer, Array<PrimExpr> indices,
+                              PrimExpr predicate = PrimExpr(), Span span = Span());
```

Comment:
Using `PrimExpr()` as the default would have an undefined object as the predicate.  While this is legal, it is rather unexpected.  While changing `PrimExpr` to be non-nullable would involve more breakage than I'd want to handle at the moment, we should avoid introducing additional cases where `PrimExpr` contains null.

Changing this to `Optional<PrimExpr> predicate = NullOpt` would avoid this issue.